### PR TITLE
enable coupling via renv

### PR DESCRIPTION
**Pull request by pfuehrlich-pik**
## Purpose of this PR
Enables the remind magpie coupling to be run using renv instead of snapshots

## Type of change

(Make sure to delete from the Type-of-change list the items not relevant to your PR)

- [ ] Bug fix 
- [ ] Refactoring
- [x] New feature 
- [ ] Minor change (default scenarios show only small differences)
- [ ] Fundamental change
- [ ] This change requires a documentation update

## Checklist:

- [x] My code follows the [coding etiquette](https://github.com/remindmodel/remind/blob/develop/main.gms#L80)
- [x] I performed a self-review of my own code
- [x] I explained my changes within the PR, particularly in hard-to-understand areas
- [x] I checked that the [in-code documentation](https://github.com/remindmodel/remind/blob/develop/main.gms#L120) is up-to-date
- [x] I adjusted the reporting in [`remind2`](https://github.com/pik-piam/remind2) where it was needed
- [x] I adjusted `forbiddenColumnNames` in [readCheckScenarioConfig.R](https://github.com/remindmodel/remind/blob/develop/scripts/start/readCheckScenarioConfig.R) in case the PR leads to deprecated switches
- [ ] All automated model tests pass (`FAIL 0` in the output of `make test`)

## Further information (optional):

* Test runs are here: 
* Comparison of results (what changes by this PR?):

### CODE REVIEW

File: start_bundle_coupled.R [hunk 1]
```diff
@@ -613,8 +618,6 @@ for (scen in common) {
 }
 
 if (! "--test" %in% flags && ! "--gamscompile" %in% flags) {
-  system(paste("cp", file.path(path_remind, ".Rprofile "), file.path(path_magpie, ".Rprofile")))
```
Comment (orichters) [inline start_bundle_coupled.R:616]: Should be kept but only executed if `is.null(renv::project())` in order to keep snapshots running.

File: start_bundle_coupled.R [hunk 2]
```diff
@@ -152,9 +158,8 @@ message("run_compareScenarios:  ", run_compareScenarios)
 if (! file.exists("output")) dir.create("output")
 
 # Check if dependencies for a REMIND model run are fulfilled
-# Use ensureRequirementsInstalled(rerunPrompt="start_bundle_coupled.R") when coupled runs are using renv.
```
Comment (orichters) [inline start_bundle_coupled.R:155]: I would maybe make that conditional on `is.null(renv::project())`. Or are snapshots discouraged in the sense "don't use that at all"? Then, line 8 should not be a warning, but a `stop`

**Review by orichters — COMMENTED** [review]
Some first comments…

We should decide whether snapshots should still be an option to be used, or whether we kick them out completely. Given that it seems to require just some few `if` statements to keep this possibility for a while, we should maybe do that? Or at least ask in the REMIND meeting if everyone is fine with renv. Personally, I probably will never go back to snapshots if I can avoid it.

If we kick snapshots out completely, also the `.snapshot.Rprofile` can be deleted.

Tutorials still need to be updated.

**Comment by pfuehrlich-pik** [comment]
I agree we should keep snapshots for now, I'll adjust as you suggested. Thanks for the review!

**Review by orichters — APPROVED** [review]
Assuming you have tested that and it works, I'm fine with the changes you made.

MAgPIE seems to take the `cfg_mag$coupling_renv_lock` and base its renv on that. So if I start a run, the new renv will be used by MAgPIE, it seems. That all seems very plausible ;)

**Comment by pfuehrlich-pik** [comment]
I updated the renv tutorial. With "first MAgPIE version that supports it" you mean a release, right?

**Comment by orichters** [comment]
Can you update [Tutorial 04 on REMIND-MAgPIE](https://github.com/remindmodel/remind/blob/develop/tutorials/04_RunningREMINDandMAgPIE.md#create-snapshot-of-r-libraries), please? I think, basically deleting the two paragraphs will do it, or just stating that `renv` will automatically detect the dependencies of MAgPIE and REMIND and install the necessary packages, or something like that.

**Comment by orichters** [comment]
> I updated the renv tutorial. With "first MAgPIE version that supports it" you mean a release, right?

Release or commits that I could cherry-pick, if that is possible.

**Review by dklein-pik — APPROVED** [review]

**Pull request merged**